pkg/sql/plan/base_binder.go

```diff
@@ -1059,7 +1067,12 @@ func (b *baseBinder) bindNumVal(astExpr *tree.NumVal, typ *Type) (*Expr, error)
 			},
 		}, nil
 	case tree.P_hexnum:
-		return returnDecimalExpr(astExpr.String())
+		s := astExpr.String()[2:]
+		if len(s)%2 != 0 {
+			s = string('0') + s
+		}
+		bytes, _ := hex.DecodeString(s)
+		return returnDecimalExpr(string(bytes), IsBin)
```

Comment:
why still decimal? you need to consider issue #5346

Reply:
I have considered it, there will be correct, please see the codes in detail. The old code is an error, for 0xXXX, if it's out of range, it will be converted into '0xXXX', so  I have done something to deal with that. And for #5346, this is the result of my local machine.

mysql> select hex(0x01020304FFFFFFF00000000CAACB00000);
+------------------------------------------+
| hex(0x01020304fffffff00000000caacb00000) |
+------------------------------------------+
| 001020304fffffff00000000caacb00000       |
+------------------------------------------+
1 row in set (0.57 sec)